File: frontend/src/types/synapse.ts
export interface ChartConfig {
  type: 'bar' | 'line' | 'donut';
  x_axis: (string | number)[];
  y_axis: number[];
  metrics_label: string;
}

export interface RecommendedAction {
  action: string;
  owner: 'medios' | 'planning' | 'estrategia';
  horizon: '24h' | '7d' | '30d';
  expected_impact: string;
  priority_score: number;
}

export interface DecisionMeta {
  intent: string;
  confidence_score: number;
  data_freshness: string;
  guardrails: string[];
  comparisons: Record<string, any>;
  actions: RecommendedAction[];
}

export interface SynapseResponse {
  response_id: string;
  narrative: string;
  render_type: 'text' | 'chart' | 'table';
  chart_config?: ChartConfig;
  raw_data?: any[];
  decision_meta?: DecisionMeta;
  /** Presente si el backend usa modo Cortex Analyst */
  cortex_analyst?: Record<string, unknown>;
  /** Reenviar en el siguiente ask para continuar el hilo Cortex Agent */
  conversation_id?: string;
}

File: backend/models/synapse.py
from pydantic import BaseModel
from typing import List, Dict, Optional, Any

class ChartConfig(BaseModel):
    type: str  # "bar", "line", "donut"
    x_axis: List[Any]
    y_axis: List[float]
    metrics_label: str


class RecommendedAction(BaseModel):
    action: str
    owner: str  # "medios" | "planning" | "estrategia"
    horizon: str  # "24h" | "7d" | "30d"
    expected_impact: str
    priority_score: float


class DecisionMeta(BaseModel):
    intent: str
    confidence_score: float
    data_freshness: str
    guardrails: List[str]
    comparisons: Dict[str, Any]
    actions: List[RecommendedAction]

class SynapseResponse(BaseModel):
    response_id: str
    narrative: str
    render_type: str  # "text", "chart", "table"
    chart_config: Optional[ChartConfig] = None
    raw_data: Optional[List[Dict]] = None
    decision_meta: Optional[DecisionMeta] = None
    # Metadatos opcionales cuando SYNAPSE_QUERY_MODE=cortex_analyst (SQL generado, avisos, etc.)
    cortex_analyst: Optional[Dict[str, Any]] = None
